Unable to compile with CUDA 11.0

`make -j _k2` has the following error:

/usr/local/cuda-11.0/bin/../targets/x86_64-linux/include/thrust/system/cuda/config.h:78:2: error: #error The version of CUB in your include path is not compatible with this release of Thrust. CUB is now included in the CUDA Toolkit, so you no longer need to use your own checkout of CUB. Define THRUST_IGNORE_CUB_VERSION_CHECK to ignore this.

Maybe can be solved by adding some "if else" to conditionally download CUB according to CUDA version in k2?

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -23,10 +23,13 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-16.04, ubuntu-18.04]
-        cuda: ["10.1", "10.2"]
+        cuda: ["10.1", "10.2", "11.0"] # 11.0 is for PyTorch 1.7.0 only
         gcc: ["5"]
         torch: ["1.6.0", "1.7.0"]
         python-version: [3.6, 3.7, 3.8]
+        exclude:
+          - cuda: "11.0"
+            torch: "1.6.0"
 
     steps:
       # refer to https://github.com/actions/checkout

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -133,7 +133,12 @@ if(K2_USE_PYTORCH)
   add_definitions(-DK2_USE_PYTORCH)
   include(torch)
 endif()
-include(cub)
+
+if(CUDA_VERSION VERSION_LESS 11.0)
+  # CUB is included in CUDA toolkit 11.0 and above
+  include(cub)
+endif()
+
 include(moderngpu)
 include(googletest)
 

--- a/docs/source/installation.rst
+++ b/docs/source/installation.rst
@@ -104,11 +104,11 @@ Install k2 from source
 
 Before compiling k2, some preparation work has to be done:
 
-- Have a compiler supporting at least c++14, e.g., GCC >= 5.0, Clang >= 3.4.
+- Have a compiler supporting at least C++14, e.g., GCC >= 5.0, Clang >= 3.4.
 - Install CMake. CMake 3.11.0 and 3.18.0 are known to work.
 - Install Python3. Python 3.6, 3.7 and 3.8 are known to work.
 - Install PyTorch. PyTorch 1.6 and 1.7 are known to work.
-- Install CUDA toolkit. CUDA 10.1 and 10.2 are known to work.
+- Install CUDA toolkit. CUDA 10.1, 10.2, and 11.0 are known to work.
 - Install cuDNN. Please install a version that is compatible with the
   CUDA toolkit you are using.
 

--- a/k2/csrc/CMakeLists.txt
+++ b/k2/csrc/CMakeLists.txt
@@ -64,7 +64,9 @@ set_target_properties(context PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
 set_target_properties(context PROPERTIES OUTPUT_NAME "k2context")
 
 # lib deps
-target_link_libraries(context PUBLIC cub)
+if(CUDA_VERSION VERSION_LESS 11.0)
+  target_link_libraries(context PUBLIC cub)
+endif()
 target_link_libraries(context PUBLIC fsa)
 target_link_libraries(context PUBLIC k2_log)
 target_link_libraries(context PUBLIC k2_nvtx)

--- a/scripts/github_actions/install_cuda.sh
+++ b/scripts/github_actions/install_cuda.sh
@@ -20,6 +20,10 @@ case "$cuda" in
     url=http://developer.download.nvidia.com/compute/cuda/10.2/Prod/local_installers/cuda_10.2.89_440.33.01_linux.run
     filename=cuda_10.2.89_440.33.01_linux.run
     ;;
+  11.0)
+    url=http://developer.download.nvidia.com/compute/cuda/11.0.2/local_installers/cuda_11.0.2_450.51.05_linux.run
+    filename=cuda_11.0.2_450.51.05_linux.run
+    ;;
   *)
     echo "Unknown cuda version: $cuda"
     exit 1

--- a/scripts/github_actions/install_cudnn.sh
+++ b/scripts/github_actions/install_cudnn.sh
@@ -15,6 +15,10 @@ case $cuda in
     filename=cudnn-10.2-linux-x64-v8.0.2.39.tgz
     url=http://www.mediafire.com/file/sc2nvbtyg0f7ien/cudnn-10.2-linux-x64-v8.0.2.39.tgz/file
     ;;
+  11.0)
+    filename=cudnn-11.0-linux-x64-v8.0.5.39.tgz
+    url=http://www.mediafire.com/file/sc2nvbtyg0f7ien/cudnn-11.0-linux-x64-v8.0.5.39.tgz/file
+    ;;
   *)
     echo "Unsupported cuda version: $cuda"
     exit 1

--- a/scripts/github_actions/install_cudnn.sh
+++ b/scripts/github_actions/install_cudnn.sh
@@ -17,7 +17,7 @@ case $cuda in
     ;;
   11.0)
     filename=cudnn-11.0-linux-x64-v8.0.5.39.tgz
-    url=http://www.mediafire.com/file/sc2nvbtyg0f7ien/cudnn-11.0-linux-x64-v8.0.5.39.tgz/file
+    url=https://www.mediafire.com/file/abyhnls106ko9kp/cudnn-11.0-linux-x64-v8.0.5.39.tgz/file
     ;;
   *)
     echo "Unsupported cuda version: $cuda"

--- a/scripts/github_actions/install_torch.sh
+++ b/scripts/github_actions/install_torch.sh
@@ -33,10 +33,14 @@ case ${torch} in
         package="torch==1.7.0"
         url=
         ;;
+      11.0)
+        package="torch==1.7.0+cu110"
+        url=https://download.pytorch.org/whl/torch_stable.html
+        ;;
     esac
     ;;
   *)
-    echo "Unsupported Pytorch version: ${torch}"
+    echo "Unsupported PyTorch version: ${torch}"
     exit 1
     ;;
 esac

